### ISSUE TITLE
Is it possible to launch BAppTabBarController on top of a pre-existing tabbarcontroller?

### ISSUE THREAD

**Issue by Absentia** [chat-sdk/chat-sdk-ios#79]
I have an app with 4 tabs.  The fourth tab has a "Open Chat" button that launches the Chat SDK with the UI copied over from the demo project.  So, essentially I am launching a tabBarController on top of another tabBarController.  For the most part it works great, but there are a few issues that make me wonder whether or not the SDK is incompatible with my current project.

You can see from this screenshot that everything looks great when I first launch the SDK - all of the tabs load fine and I can click through everything without issues - https://www.dropbox.com/s/wgpvfjm3fs8txzv/Simulator%20Screen%20Shot%20Jun%2010%2C%202017%2C%2012.32.39%20PM.png?dl=0

When the project seems to get confused is when I launch a view within the controller, such as tapping on a chat room or entering a conversation thread.  The view will load fine, but when I back out, the first tab disappears.  Here's a screenshot of what I mean -https://www.dropbox.com/s/r68dkjw8goif3au/Simulator%20Screen%20Shot%20Jun%2010%2C%202017%2C%2012.33.12%20PM.png?dl=0

Finally, after that first tab vanishes and I attempt to log out via the "Profile" page, the app crashes with the following error - "Application tried to present modally an active controller <BAppTabBarController"

So basically, I've followed the instructions to launch the SDK with the following code:
UIViewController *chatSDK = [[BAppTabBarController alloc] initWithNibName:Nil bundle:Nil];
[self presentViewController:chatSDK animated:YES completion:nil];

My question is, is there another way of launching it that would allow the two tabbarcontrollers to operate independently of each other?  If this is too complex of a question feel free to close it, but I've been searching stack overflow for two days straight and I've seriously hit a wall with this one so I figured I would throw it out there.

**Comment by bensmiley** [comment]
I think that this isn't a problem with the Chat SDK - it's rather a problem with launching one set of tabs over another set of tabs which is a iOS UI issue. 

Without investigating further, I couldn't say if this is possible or not. The issue where the tab icon disappears is very strange. I'm sure that iOS is not designed to have multiple tab bars running in parallel (but that's not to say it's not possible). 

From [Apple's human interface guidelines](http://developer.apple.com/library/ios/#DOCUMENTATION/UserExperience/Conceptual/MobileHIG/UIElementGuidelines/UIElementGuidelines.html):

> Use a tab bar to give users access to different perspectives on the same set of data or different subtasks related to the overall function of your app. When you use a tab bar, follow these guidelines:

>Don’t use a tab bar to give users controls that act on elements in the current mode or screen. If you need to provide controls for your users, use a toolbar instead (for usage guidelines, see “Toolbar”).

>In general, use a tab bar to organize information at the application level. A tab bar is well-suited for use in the main app view because it’s a good way to flatten your information hierarchy and provide access to several peer information categories or modes at one time.

The log out issue is easier to explain, when you logout the app will attempt to present the login screen. If the whole app is a modal view, this will fail. 

Personally, I'd recommend a different approach. You could have the chat in a side menu like [this](https://github.com/aryaxt/iOS-Slide-Menu). It would be pretty easy to setup and would provide a better user experience. 

If you wanted to try to implement this and then make it available to the community, I'd be happy to offer free support.

**Comment by Absentia** [comment]
Ok, thanks for the link.  I'll definitely look into that

**Comment by Absentia** [comment]
Looks like I discovered what was causing both issues that I mentioned in my first post.  BChatViewController.m has the following line in the ViewWillAppear method - 

 UIViewController * mainViewController = [[BAppTabBarController alloc] initWithNibName:Nil bundle:Nil];

Deleting that line has resolved both issues.  I've been testing the SDK for a while since deleting it and the deletion of that line doesn't seem to have caused any further issues, but you way be more aware of it's importance than I am.
